packages/firestore/src/core/sync_engine.ts

```diff
@@ -404,11 +406,32 @@ export class SyncEngine implements RemoteSyncer {
     }
   }
 
+  // PORTING NOTE: Multi-tab only
+  applyPendingBatch(batchId: BatchId): Promise<void> {
+    // TODO(multitab): Implement applyPendingBatch
+    throw new FirestoreError(
+      Code.UNIMPLEMENTED,
+      'applyPendingBatch not implemented'
+    );
+  }
+
+  applySuccessfulWrite(mutationBatchResult: MutationBatchResult): Promise<void>;
+  // PORTING NOTE: Multi-tab only
+  applySuccessfulWrite(batchId: BatchId): Promise<void>;
```

Comment:
This was intentional to keep the public interface of SyncEngine manageable (one could argue that adding an overload also doesn't help here). While there is not much in common between the two versions of `applySuccessfulWrite`, `rejectFailedWrite` on the other hand can probably be used as is.

I am totally open to changing the semantics here - do you have some suggestions for naming? `applyAcknowledgedWrite` comes to mind. 

Note that this callback will still need to resolve the promises for writes if a write gets issued from a secondary tab. Otherwise, we could consider merging with `applyPendingBatch`.